src/cart/crt_group.c

```diff
@@ -1379,7 +1379,7 @@ crt_group_version_set(crt_group_t *grp, uint32_t version)
 		D_GOTO(out, rc = -DER_INVAL);
 	}
 
-	D_RWLOCK_RDLOCK(&grp_priv->gp_rwlock);
+	D_RWLOCK_WRLOCK(&grp_priv->gp_rwlock);
```

Comment:
this seems like the real fix here which could potentially use a compare and swap.  The read lock to read one field doesn't accomplish much outside of disallowing the simultaneous modification of said field.